crates/router/src/core/payments.rs

```diff
@@ -615,14 +615,19 @@ async fn decide_payment_method_tokenize_action(
                 connector_name
             );
 
-            match redis_conn.get_key::<String>(&key).await {
-                Ok(connector_token) => TokenizationAction::ConnectorToken(connector_token),
-                Err(error) => {
-                    logger::debug!(connector_token_redis_error=?error);
+            let connector_token_option = redis_conn
+                .get_key::<Option<String>>(&key)
+                .await
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Failed to fetch the token from redis")?;
+
+            match connector_token_option {
+                Some(connector_token) => Ok(TokenizationAction::ConnectorToken(connector_token)),
+                None => {
                     if is_connector_tokenization_enabled {
-                        TokenizationAction::TokenizeInConnector
+                        Ok(TokenizationAction::TokenizeInConnector)
                     } else {
-                        TokenizationAction::SkipConnectorTokenization
+                        Ok(TokenizationAction::TokenizeInRouter)
                     }
                 }
             }
```

Comment:
Hey, minor nit pick, now that you are using a variable you can do something like this:
```rust
match (connector_token_option, is_connector_tokenization_enabled) {
    (Some(connector_token), _) => Ok(TokenizationAction::ConnectorToken(connector_token)),
    (None, true) => Ok(TokenizationAction::TokenizeInConnector),
    (None, false) => Ok(TokenizationAction::TokenizeInRouter)
}
```

Reply:
can change this, moving forward